### PR TITLE
Update remove AbstractWindows And Friends

### PR DESCRIPTION
**Pull request by aneury1**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,66 +11,70 @@ endif()
...
+set(CMAKE_CXX_STANDARD 14)
 
-set(SDL2IMAGE_INSTALL OFF)
-set(BUILD_SHARED_LIBS FALSE)
-set(SDL2_DISABLE_INSTALL OFF)
-set(SDL2TTF_VENDORED TRUE)
+# set(SDL2IMAGE_INSTALL OFF)
+# set(BUILD_SHARED_LIBS FALSE)
+# set(SDL2_DISABLE_INSTALL OFF)
+# set(SDL2TTF_VENDORED TRUE)
```
Comment (victordmontero) [inline CMakeLists.txt:20]:
```suggestion
```
No need to commit them commented

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,66 +11,70 @@ endif()
 
 # set(CMAKE_EXPORT_COMPILE_COMMANDS true)
 
-set(CMAKE_BUILD_TYPE Debug)
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_BUILD_TYPE Release)
```
Comment (victordmontero) [inline CMakeLists.txt:14]: I don't think we have to fix the build type as release. I assume that while developing you rarely build as release. I think it is better to leave it as it was.

File: CMakeLists.txt [hunk 3]
```diff
@@ -11,66 +11,70 @@ endif()
...
+# set(SDL2_DISABLE_INSTALL OFF)
+# set(SDL2TTF_VENDORED TRUE)
 
 if(DOENGINE_TESTS)
     enable_testing()
     add_subdirectory(test)
 endif()
 
-include(FetchContent)
+# include(FetchContent)
```
Comment (victordmontero) [inline CMakeLists.txt:27]:
```suggestion
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -11,66 +11,70 @@ endif()
...
+
+# # SDL2_mixer
+# SET(SDL2MIXER_VENDORED TRUE)
+# FetchContent_Declare(
+#     sdl_mixer
+#     GIT_REPOSITORY https://github.com/libsdl-org/SDL_mixer.git
+#     GIT_TAG release-2.6.3
+# )
+
+# FetchContent_MakeAvailable(sdl_mixer)
```
Comment (victordmontero) [inline CMakeLists.txt:68]:
```suggestion
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -101,33 +105,21 @@ add_library(${PROJECT_NAME}
...
-    sample/SortedState.cpp
-    sample/main.cpp
-)
+# target_link_options(${PROJECT_NAME} PRIVATE
+#     "LINKER:-static" -static-libgcc -static-libstdc++)
+# target_link_libraries(${PROJECT_NAME} PRIVATE
+#     SDL2::SDL2
+#     SDL2::SDL2main
+#     SDL2_ttf SDL2_image
+#     SDL2_mixer)
```
Comment (victordmontero) [inline CMakeLists.txt:117]:
```suggestion
```

File: includes/v2d_vector.h [hunk 6]
```diff
@@ -43,5 +43,5 @@ template<class T> inline v2d_generic<T> operator / (const int& lhs, const v2d_ge
 typedef v2d_generic<int> iVector2;
 typedef v2d_generic<float> fVector2;
 typedef v2d_generic<double> dVector2;
-typedef iVector2 Point;
+//typedef iVector2 Point;
```
Comment (victordmontero) [inline includes/v2d_vector.h:46]: ??

File: src/Drawable/Canvas.cpp [hunk 7]
```diff
@@ -201,40 +201,40 @@ void arc(Window *window, int x, int y, int radius, double startAngle, double end
...
  
     }
  
 }
 
-void DrawPoint(const Point &point, const Color& color, doengine::AbstractWindow *window){
-   /// SDL_SetRenderDrawColor(window->getRender(), color.r, color.g, color.b, color.a);
-    SDL_RenderDrawPoint(static_cast<SDL_Renderer*>(window->getRender()->getNativeRenderer()), point.x, point.y);
+void DrawPoint(const Point &point, const Color& color){
+   /// SDL_SetRenderDrawColor(Application::getApplication()->getRender(), color.r, color.g, color.b, color.a);
```
Comment (victordmontero) [inline src/Drawable/Canvas.cpp:219]:
```suggestion
 ```

File: src/Drawable/Canvas.cpp [hunk 8]
```diff
@@ -201,40 +201,40 @@ void arc(Window *window, int x, int y, int radius, double startAngle, double end
...
+    SDL_RenderDrawPoint(static_cast<SDL_Renderer*>(Application::getApplication()->getRender()->getNativeRenderer()), point.x, point.y);
 }
 void DrawLine(const Point& p1, const Point& p2, const Color &color, Window *window){
-    /// SDL_SetRenderDrawColor(window->getRender(), color.r, color.g, color.b, color.a);
-     SDL_RenderDrawLine(static_cast<SDL_Renderer*>(window->getRender()->getNativeRenderer()), p1.x, p1.y, p2.x, p2.y);
+    /// SDL_SetRenderDrawColor(Application::getApplication()->getRender(), color.r, color.g, color.b, color.a);
```
Comment (victordmontero) [inline src/Drawable/Canvas.cpp:223]:
```suggestion
```

File: src/Drawable/Canvas.cpp [hunk 9]
```diff
@@ -201,40 +201,40 @@ void arc(Window *window, int x, int y, int radius, double startAngle, double end
...
 void DrawRect(const Rect &rect, const Color& color, Window *window){
- ///  SDL_SetRenderDrawColor(window->getRender(), color.r, color.g, color.b, color.a);
-   SDL_RenderDrawRect(static_cast<SDL_Renderer*>(window->getRender()->getNativeRenderer()), &rect);
+ ///  SDL_SetRenderDrawColor(Application::getApplication()->getRender(), color.r, color.g, color.b, color.a);
+   SDL_RenderDrawRect(static_cast<SDL_Renderer*>(Application::getApplication()->getRender()->getNativeRenderer()), &rect);
 }
 void DrawFillRect(const Rect &rect, const Color& color, Window *window){
-  ///SDL_SetRenderDrawColor(window->getRender(), color.r, color.g, color.b, color.a);
-  SDL_RenderFillRect(static_cast<SDL_Renderer*>(window->getRender()->getNativeRenderer()), &rect);
+  ///SDL_SetRenderDrawColor(Application::getApplication()->getRender(), color.r, color.g, color.b, color.a);
```
Comment (victordmontero) [inline src/Drawable/Canvas.cpp:232]:
```suggestion
```

File: src/Drawable/Canvas.cpp [hunk 10]
```diff
@@ -364,35 +364,35 @@ void FillCircle(int x, int y, int radius, const Color& color, doengine::Abstract
...
-   DrawSinglePoint(offset, color, window);
+   DrawSinglePoint(offset, color);
 }
 
 
-void CanvasTextDrawerCommand::Draw(doengine::AbstractWindow *window){
+void CanvasTextDrawerCommand::Draw(){
  
-   ///SDL_SetRenderDrawColor(window->getRender(), color.r, color.g, color.b, color.a);
+   ///SDL_SetRenderDrawColor(Application::getApplication()->getRender(), color.r, color.g, color.b, color.a);
```
Comment (victordmontero) [inline src/Drawable/Canvas.cpp:393]:
```suggestion
```

File: src/Drawable/Canvas.cpp (hunk 8 above)
Comment (aneury1) [reply src/Drawable/Canvas.cpp:223]: Im removing no commenting.

**Pull request closed without merging**